### DISCUSSION TITLE
support user profile picture

### DISCUSSION THREAD

**Discussion by helmut72** [keycloak/keycloak#15943] [Ideas]
It would be great if Keycloak had a feature for uploading and storing a user profile image, that is used as source url in OIDC and SAML2.

If the user base is LDAP, `jpegPhoto` attribute should be used to read the user profile picture and should also save back, if LDAP is writeable.

In 2022 some apps are using user profile pictures, for example Nextcloud. It make only sense to ask for user profile picture support in other apps, if the IDP supports it at all.

Google and Microsoft support user profile picture for a longer time.

Also users are lazy to upload profile pictures in every single app and it looks a lot nicer and more personal with a profile picture instead of a placeholder icon.

Thank you.

**Comment by ssilvert** [comment]
It's part of the plan. 

See https://github.com/keycloak/keycloak-ui/issues/3311

**Reply by helmut72** [reply]
Thank you, looks like what I've asked for.

**Reply by helmut72** [reply]
Time goes by and I found account3 and declarative-user-profile options. Do you mean this with "the plan"?

This means I need somewhere an image server to provide user profile picture and the url can be assembled via user profile in Keycloak?

Keycloak doesn't provide any feature, where the users can add their own images and Keycloak serve the pictures? But this also means that the users need at least two places to manage their account? For example Keycloak to manage password and tokens and somewhere else to add an own profile picture?

**Reply by exhuma** [reply]
The link is down :(

**Reply by helmut72** [reply]
What I remember it linked to "Project Masthead":

https://github.com/keycloak/keycloak/issues/17746

But this doesn't include a webserver component to store Profile Pictures. Zitadel IDP supports Profile Picture out of the box, but lacks with LDAP integration.
